app/stac_api/models.py

```diff
@@ -197,27 +190,469 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
+        :param old_start_datetime: item's old value for properties_start_datetime
+        :param new_start_datetime: item's updated value for properties_start_datetime
+        :param old_end_datetime: item's old value for properties_end_datetime
+        :param new_end_datetime: item's updated value for properties_end_datetime
+        :param item_id: the id of the item being treated
+        :param qs: (optional) queryset with all items of the collection excluding the one
+        being updated.
+        '''
+        logger.debug(
+            'Updating item %s in collection (adjusting end_datetime) %s',
+            item_id,
+            self.collection_name
+        )
+        if old_end_datetime != new_end_datetime:
+            # end_datetime was updated
+            if old_end_datetime == self.cache_end_datetime:
+                # item's old end_datetime was defining the right bound of
+                # the collection's temporal extent interval before update
+                if new_end_datetime > old_end_datetime:
+                    # item's end_datetime was shifted to the right (later)
+                    self.cache_end_datetime = new_end_datetime
+                else:
+                    # item's end_datetime was shifted to the left (earlier)
+                    # but was defining the right bound of the collection's
+                    # temporal extent.
+                    # --> hence the new end_datetime of the collection needs
+                    # to be determined:
+
+                    # get latest end_datetime or none, in case none exists
+                    # this should be the case, if the item that is currently
+                    # being updated is the only item in the collection or
+                    # no item has a range defined (no start_ and end_datetimes)
+                    # but all do only have properties.datetimes defined.
+                    if bool(qs.filter(properties_end_datetime__isnull=False)):
+                        latest_end_datetime = qs.filter(
+                            properties_end_datetime__isnull=False
+                        ).latest('properties_end_datetime').properties_end_datetime
+                    else:
+                        latest_end_datetime = None
+                    # get latest datetime, or none in case none exists
+                    # this should be the case, if the item that is currently
+                    # being updated is the only item in the collection or
+                    # no item has properties.datetime defined but all do have
+                    # only ranged (start_ and end_datetimes defined)
+                    if bool(qs.filter(properties_datetime__isnull=False)):
+                        latest_datetime = qs.filter(
+                            properties_datetime__isnull=False
+                        ).latest('properties_datetime').properties_datetime
+                    else:
+                        latest_datetime = None
```

Comment:
Same as above this could be regular function: `latest_datetime, latest_start_datetime = get_latest_datetime(queryset)

Reply:
see above and below